### PR TITLE
[move-cli] Only print differences in changed resources during verbose…

### PR DESCRIPTION
**Pull request by DreamLinuxer**
## Motivation

Issue https://github.com/diem/diem/issues/8545

## Test Plan
Screen Shot 2021-11-13 at 6 18 27 PM

### CODE REVIEW

File: language/tools/move-cli/src/sandbox/utils/mod.rs [hunk 1]
```diff
@@ -98,6 +100,41 @@ fn print_struct_with_indent(value: &AnnotatedMoveStruct, indent: u64) {
...
+
+    let Changeset { diffs, .. } = Changeset::new(&prev_str, &new_str, "\n");
+
+    let mut t = term::stdout().unwrap();
+
+    for diff in diffs {
+        match diff {
+            Difference::Same(ref x) => {
+                t.reset().unwrap();
+                let _ = writeln!(t, " {}{}", indent_str, x);
```
Comment (tzakian) [inline language/tools/move-cli/src/sandbox/utils/mod.rs:121]: Nit: you should probably `unwrap` these to make sure that the result is used
```suggestion
            writeln!(t, " {}{}", indent_str, x).unwrap();
```

File: language/tools/move-cli/src/sandbox/utils/mod.rs [hunk 2]
```diff
@@ -98,6 +100,41 @@ fn print_struct_with_indent(value: &AnnotatedMoveStruct, indent: u64) {
...
+    value2: &AnnotatedMoveStruct,
+    indent: u64,
+) {
+    let indent_str: String = (0..indent).map(|_| " ").collect::<String>();
+    let prev_str = format!("{}", value1);
+    let new_str = format!("{}", value2);
+
+    let Changeset { diffs, .. } = Changeset::new(&prev_str, &new_str, "\n");
+
+    let mut t = term::stdout().unwrap();
```
Comment (tzakian) [inline language/tools/move-cli/src/sandbox/utils/mod.rs:115]: It feels like using `term` for this is a bit heavyweight. We normally use something like the [colored](https://docs.rs/colored/2.0.0/colored/) crate to output colored text, but I'm fine with using this as well.

**Review by tzakian — COMMENTED** [review]
This looks great to me! Thanks for doing this :) A couple minor little things to change, but otherwise should be ready to go in

**Review by tzakian — APPROVED** [review]
LGTM! Thanks for this!

**Review by tzakian — APPROVED** [review]

**Comment by DreamLinuxer** [comment]
\land

**Review by tzakian — DISMISSED** [review]
/land

**Pull request merged**